[DOCS-8379] Add watchdog terminations docs, clean app hangs one, and update corresponding mobile vitals (#24020)
DataDog/documentation: content/en/real_user_monitoring/error_tracking/mobile/ios.md

                          "metric": "@session.time_spent",
                          "interval": 3600000
                      },
                      "storage": "hot"
                  }
              ],
              "response_format": "timeseries",
              "style": {
                  "palette": "dog_classic",
                  "order_by": "values",
                  "line_type": "solid",
                  "line_width": "normal"
              },
              "display_type": "line"
          }
      ],
      "yaxis": {
          "include_zero": true,
          "scale": "sqrt"
      },
      "markers": [
          {
              "value": "y > 12000000000",
              "display_type": "error dashed"
          },
          {
              "value": "6000000000 < y < 12000000000",
              "display_type": "warning dashed"
          },
          {
              "value": "0 < y < 6000000000",
              "display_type": "ok dashed"
          }
      ]
  }
  ```
</details>

#### Disable app hang monitoring

To disable app hang monitoring, update the initialization snippet and set the `appHangThreshold` parameter to `nil`.

> ### Add watchdog terminations reporting

Reviewer: ```suggestion
### Add Watchdog Terminations reporting
```

Author: We've been following a lowercase format for app hangs already, so to remain consistent I used the same format for watchdog terminations

In the Apple ecosystem, the operating system employs a watchdog mechanism to monitor the health of applications, and terminates them if they become unresponsive or consume excessive resources like CPU, memory, etc. These Watchdog Terminations are fatal and not recoverable (more details in the official [Apple documentation][12]).

By default, watchdog terminations reporting is **disabled**, but you can enable it by using the `trackWatchdogTerminations` initialization parameter.

Watchdog terminations are reported through the RUM iOS SDK only (not through [Logs][4]).

When enabled, a watchdog termination will be reported and attached to the previous RUM Session on the next application launch, based on heuristics:

- The application was not upgraded in the meantime,

> - And it did id not call neither `exit`, nor `abort`,

Reviewer: ```suggestion
- And it did not call neither `exit`, nor `abort`,
```

- And it did not crash, either because of an exception, or because of a fatal [app hang][13],
- And it was not force-quitted by the user,
- And the device did not reboot (which includes upgrades of the operating system).

{{< img src="real_user_monitoring/error_tracking/ios-watchdog-termination.png" alt="A watchdog termination in the RUM Error side panel." style="width:60%;" >}}

> #### Enable watchdog terminations reporting

Reviewer: ```suggestion
#### Enable Watchdog Terminations reporting
```

> #### Enable watchdog terminations reporting

Author: Thanks, updated

- The SDK implements a secondary thread for monitoring app hangs. To reduce CPU utilization, it tracks hangs with a tolerance of 2.5%, which means some hangs that last close to the `appHangThreshold` may not be reported.

#### Compute the hang rate of your application

[Xcode Organizer][15] and [MetricKit][16] both provide a hang rate metric defined as "the number of seconds per hour that the app is unresponsive, while only counting periods of unresponsiveness of more than 250 ms."

> To compute a similar hang rate on Datadog, make sure:
> 
> 1. That app hang reporting is enabled
> 
> 2. That the app hang threshold is equal or below 250 ms
> 
> 3. That the `@error.category` and `@freeze.duration` attribute reported on your app hangs errors in RUM are avaible in your facets (this should be the case by default. If it's not, you can manually [create facets][17])
> 
> If all these pre-requisites are met, then create a new [Timeseries widget][18] on a Dashboard or a Notebook, and paste the following snippet in the JSON tab of your widget, under the "Graph your data" section

Reviewer 2: ```suggestion
To compute a similar hang rate on Datadog, make sure:

1. That app hang reporting is enabled.
2. That the app hang threshold is equal or below 250 ms.
3. That the `@error.category` and `@freeze.duration` attribute reported on your app hangs errors in RUM are available in your facets (this should be the case by default. If it's not, you can manually [create facets][17]).

If all these prerequisites are met, then create a new [Timeseries widget][18] on a Dashboard or a Notebook, and paste the following snippet in the JSON tab of your widget, under the "Graph your data" section:
```

Watchdog terminations are reported through the RUM iOS SDK only (not through [Logs][4]).

When enabled, a watchdog termination will be reported and attached to the previous RUM Session on the next application launch, based on heuristics:

- The application was not upgraded in the meantime,

- And it did not call neither `exit`, nor `abort`,

- And it did not crash, either because of an exception, or because of a fatal [app hang][13],

- And it was not force-quitted by the user,

- And the device did not reboot (which includes upgrades of the operating system).

{{< img src="real_user_monitoring/error_tracking/ios-watchdog-termination.png" alt="A watchdog termination in the RUM Error side panel." style="width:60%;" >}}

#### Enable watchdog terminations reporting

To enable watchdog terminations reporting:

1. [Enable Crash Reporting][19]

2. Update the initialization snippet with the `trackWatchdogTerminations` flag:

    ```swift
    RUM.enable(
        with: RUM.Configuration(
            applicationID: "<rum application id>",
            trackWatchdogTerminations: true
        )
    )
    ```

#### Troubleshoot watchdog terminations

> When an application is terminated by the iOS Watchdog, it doesn’t get any termination signal. Because of this lack of termination signal, watchdog terminations do not contain any stack trace. To troubleshoot watchdog terminations, we recommend looking at the [vitals][14] of the parent RUM View (CPU Ticks, Memory).

Reviewer 2: ```suggestion
When an application is terminated by the iOS Watchdog, it doesn’t get any termination signal. As a result of this lack of a termination signal, watchdog terminations do not contain any stack trace. To troubleshoot watchdog terminations, we recommend looking at the [vitals][14] of the parent RUM View (CPU Ticks, Memory).
```
